### DISCUSSION TITLE
Stream not loading on client side

### DISCUSSION THREAD

**Discussion by SPrasadT** [ant-media/Ant-Media-Server#5733] [Q&A]
Hi Everyone,

I have a situation, where the webRTC stream works fine on my network, and the same doesn't on a banking network, I checked the Chrome developer console logs, and see not error codes on the network tab/console for any javascript errors.

following is the code which is used in angular
```
/*stream using the updated implementation*/
  antMediaStream() {
    let websocketURL = 'wss://'+this.deviceData.webRTCServerEndpoint+'/'+this.deviceData.environment+'/websocket';
    this.url=websocketURL;
    let mediaConstraints = {
      video: true,
      audio: false
    };

    let pc_config = {
      'iceServers': [{
        'urls': 'stun:stun1.l.google.com:19302'
      }]
    };

    let sdpConstraints = {
      OfferToReceiveAudio: true,
      OfferToReceiveVideo: true
    };

     this.webRTCAdaptor = new WebRTCAdaptor({
      websocket_url: websocketURL,
      mediaConstraints: mediaConstraints,
      peerconnection_config: pc_config,
      sdp_constraints: sdpConstraints,
      remoteVideoId: "imgstream",
      isPlayMode: true,
      callback: (info: string, obj: any) => {
        if (info == "play_started") {
          console.log(info + " play_started");
          this.startPlaying()

        } else if (info == "play_finished") {
          console.log(info + " play_finished");

        }
        else {
          console.log(info + " notification received");
        }
      },
      callbackError:  (error: any, message: any)=> {
        console.log("Error Message :",error);
        //this.toastr.errorToast("Unable to connect to stream,Please try again")
      }
    });


    setTimeout(() => {
      this.startPlaying()
    }, 1000);
  }
```

I did set up my own TURN server, and used the TURN endpoint but still the issue is same. Any pointers on this would be really helpful.

Thanks.

**Comment by USAMAWIZARD** [comment]
most likely its a turn server issue can you please check if the turn server is working properly by entering your credentials [here](https://webrtc-turn-server-test.vercel.app/).
if it works the stream should appear in the detention side.
https://webrtc-turn-server-test.vercel.app/
![image](https://github.com/ant-media/Ant-Media-Server/assets/47350008/70b3762a-238a-4a72-a968-8cc0af18cdd2)

**Reply by SPrasadT** [reply]
We tested this as suggested and I still don't see the stream, just an FYI, the stream is not coming up inside a Citrix VDI environment.

![image](https://github.com/ant-media/Ant-Media-Server/assets/26184292/1d90aa8d-1f24-4e52-95f1-8a62a95959ea)

**Reply by USAMAWIZARD** [reply]
Okay that means that turn server is working , Can you please make sure you have added the Turn Configuration in the below suggested way

**Reply by punitqyrus** [reply]
i am getting the same issue where the stream is working on my machine but when i try to access the stream on  my CITRIX VDI i dont see the stream and when i tried the above test using `https://webrtc-turn-server-test.vercel.app/` within the VDI i dont see the destination.
what should i do to debug this further, does it requires some sort of whitelisting if so which URL and PORTS is to be whitelisted.

**Reply by USAMAWIZARD** [reply]
please open port 3478 and 3479 on the network

**Reply by mekya** [reply]
hi @USAMAWIZARD, @SPrasadT, @punitqyrus 

I think it's worth scheduling a meeting. It may save a good amount of time.

If it sounds good, just provide your availability for the meeting in this week.

Regards

**Reply by USAMAWIZARD** [reply]
Sure, @punitqyrus  Please schedule a meeting with me [using this link](https://calendly.com/)

**Reply by punitqyrus** [reply]
@USAMAWIZARD i have scheduled a meeting with the provided link.
Thanks.

**Reply by USAMAWIZARD** [reply]
Sure , See you then

**Comment by lastpeony** [comment]
How did you pass turn server to iceServers list? 
It should be ın this format:
```
var pc_config =
{
    'iceServers': [
        {
            'urls': 'stun:stun1.l.google.com:19302'
        },
        {
            'urls': 'TURN_IP:3478',
            'username': 'username',
            'credential': 'passsword'
        }]
}

var webRTCAdaptor = new WebRTCAdaptor({
    peerconnection_config: pc_config,
    // other options
})
```
Also ports UDP:5000-65000 (WebRTC and RTSP) should be allowed
